publish.sh

```diff
@@ -1,11 +1,80 @@
-#!/bin/bash
-timestamp=$(date -u)
-cd /home/friedger/_repos/github/friedger/pub-stacks-dump
-cd ../../psq/stacks-dump
-node report /tmp/stacks-testnet-bb8423eafa69dc8f/ -a > ../../friedger/pub-stacks-dump/stacks-dump.txt
-cd ../../friedger/pub-stacks-dump
+#!/usr/bin/env bash
+
+# This script will run stacks-dump against
+# an active stacks-node miner's data and
+# output the results to a web page that
+# is uploaded to GitHub and published via
+# GitHub Pages.
+
+########
+# INIT #
+########
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# timestamp in UTC
+__timestamp=$(date -u)
+
+# read input for loop (optional)
+__action="${1:-}"
+
+## UPDATE VARIABLES BELOW FOR YOUR SYSTEM
+
+# directory for working_dir data from stacks-node
+__stacksnode="/tmp/stacks-testnet-bb8423eafa69dc8f/"
+# directory for running stacks-dump
+__stacksdump="/home/friedger/_repos/github/psq/stacks-dump"
+# directory for repo to publish results
+__publishdir="/home/friedger/_repos/github/friedger/pub-stacks-dump"
+# file name for saving stacks-node data
+__outputfile="stacks-dump.txt"
+# website to access data after published
+__website="https://friedger.github.io/pub-stacks-dump/"
+# twitter account used for twitter card in SEO
+__twitter="@fmdroid"
+
+##########
+# SCRIPT #
+##########
+
+# Verify all directories exist before starting
+if [ ! -d "$__stacksnode" ]; then
+  printf "stacks-node working directory not found, please check the variable in the script."
+  exit
+elif [ ! -d "$__stacksdump" ]; then
+  printf "stacks-dump directory not found, please check the variable in the script or download from GitHub.\n\nhttps://github.com/psq/stacks-dump"
+  exit
+elif [ ! -d "$__publishdir" ]; then
+  printf "publishing directory not found, please check the variable in the script."
+  exit
+fi
+
+if [ "$__action" == "" ]; then 
+  publish
+  exit
+else
+  while true
+  do
+    __timestamp=$(date -u)
+    publish
+    printf "Published at: %s", "$__timestamp"
+  done
```

Comment:
Shouldn't the script sleep for an hour here?

Reply:
Yep! See f078880, I know I forget to sleep sometimes too :zzz: